nifi-nar-bundles/nifi-social-media-bundle/nifi-twitter-processors/src/main/java/org/apache/nifi/processors/twitter/ConsumeTwitter.java

```diff
@@ -0,0 +1,366 @@
+                    "https://developer.twitter.com/en/docs/twitter-api/expansions " +
+                    "for proper usage. Possible field values include: " +
+                    "author_id, referenced_tweets.id, referenced_tweets.id.author_id, entities.mentions.username, " +
+                    "attachments.poll_ids, attachments.media_keys ,in_reply_to_user_id, geo.place_id")
+            .required(false)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final Relationship REL_SUCCESS = new Relationship.Builder()
+            .name("success")
+            .description("FlowFiles containing an array of one or more Tweets")
+            .build();
+
+    private List<PropertyDescriptor> descriptors;
+    private Set<Relationship> relationships;
+
+    private TweetStreamService tweetStreamService;
+
+    private volatile BlockingQueue<String> messageQueue;
+
+    @Override
+    protected void init(ProcessorInitializationContext context) {
+        final List<PropertyDescriptor> descriptors = new ArrayList<>();
+        descriptors.add(ENDPOINT);
+        descriptors.add(BASE_PATH);
+        descriptors.add(BEARER_TOKEN);
+        descriptors.add(QUEUE_SIZE);
+        descriptors.add(BATCH_SIZE);
+        descriptors.add(BACKOFF_ATTEMPTS);
+        descriptors.add(BACKOFF_TIME);
+        descriptors.add(MAXIMUM_BACKOFF_TIME);
+        descriptors.add(CONNECT_TIMEOUT);
+        descriptors.add(READ_TIMEOUT);
+        descriptors.add(BACKFILL_MINUTES);
+        descriptors.add(TWEET_FIELDS);
+        descriptors.add(USER_FIELDS);
+        descriptors.add(MEDIA_FIELDS);
+        descriptors.add(POLL_FIELDS);
+        descriptors.add(PLACE_FIELDS);
+        descriptors.add(EXPANSIONS);
+
+        this.descriptors = Collections.unmodifiableList(descriptors);
+
+        final Set<Relationship> relationships = new HashSet<>();
+        relationships.add(REL_SUCCESS);
+        this.relationships = Collections.unmodifiableSet(relationships);
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return this.relationships;
+    }
+
+    @Override
+    public final List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return this.descriptors;
+    }
+
+
+    @OnScheduled
+    public void onScheduled(final ProcessContext context) {
+        messageQueue = new LinkedBlockingQueue<>(context.getProperty(QUEUE_SIZE).asInteger());
+
+        tweetStreamService = new TweetStreamService(context, messageQueue, getLogger());
+        tweetStreamService.setBasePath(context.getProperty(BASE_PATH).getValue());
+        final String endpointName = context.getProperty(ENDPOINT).getValue();
+        if (ENDPOINT_SAMPLE.getValue().equals(endpointName)) {
+            tweetStreamService.start(StreamEndpoint.SAMPLE_ENDPOINT);
+        } else {
+            tweetStreamService.start(StreamEndpoint.SEARCH_ENDPOINT);
+        }
```

Comment:
Instead of passing the endpoint value to `start()`, the property value can be evaluated and set in the `TweetStreamService` constructor, which will simplify this block:
```suggestion
        tweetStreamService.start();
```

Reply:
Making the changes